builder/amazon/chroot/communicator.go

```diff
@@ -86,7 +86,7 @@ func (c *Communicator) Upload(dst string, r io.Reader, fi *os.FileInfo) error {
 	return ShellCommand(cpCmd).Run()
 }
 
-func (c *Communicator) UploadDir(dst string, src string, exclude []string) error {
+func (c *Communicator) UploadDir(dst string, src string, exclude []string, overwriteRemote bool) error {
```

Comment:
Hello there thanks for opening, I see this is a WIP. And wanted to help you with a suggestion.

Since this is breaking the params of `UploadDir`, I think it could be nice to pass a ctx and an upload config struct in here :
```suggestion
func (c *Communicator) UploadDir(ctx context.Context, cfg communicator.UploadConfig) error {
```

I think it's always nice to be able to pass a context for an upload operation that could take a long time if a user would like to cancel it or something.